q2_fmt/tests/test_engraftment.py

```diff
@@ -1023,3 +1023,81 @@ def test_drop_incomplete_timepoints_list(self):
                       float("Nan")]}).set_index('id')
         metadata_df = _drop_incomplete_timepoints(metadata_df, "group", [3, 2])
         self.assertEqual(metadata_df["group"].dropna().unique(), [float(1)])
+
+    def test_rename_features_with_delim(self):
+        metadata_df = pd.DataFrame({
+                'id': ['sample1', 'sample2', 'sample3',
+                       'donor1'],
+                'Ref': ['donor1', 'donor1', 'donor1', float("Nan")],
+                'subject': ['sub1', 'sub1', 'sub1', float("Nan")],
+                'group': [1, 1, 1, float("Nan")]}).set_index('id')
+        metadata = Metadata(metadata_df)
+        table_df = pd.DataFrame({
+                'id': ['sample1', 'sample2', 'sample3',
+                       'donor1'],
+                'Feature;1': [0, 0, 1, 1],
+                'Feature;2': [0, 1, 1, 1],
+                'Feature;3': [0, 0, 1, 0]}).set_index('id')
+        feature_peds_df = feature_peds(table=table_df, metadata=metadata,
+                                       time_column="group",
+                                       reference_column="Ref",
+                                       subject_column="subject")
+        _rename_features(data=feature_peds_df, level_delimiter=";")
+        Fs1 = feature_peds_df.set_index("id").at['Feature 1',
+                                                 'subject']
+        Fs2 = feature_peds_df.set_index("id").at['Feature 2',
+                                                 'subject']
+        self.assertEqual("1", Fs1)
```

Comment:
Ok so just so I understand - expected behavior here is that the highest taxonomic resolution is what's retained, correct? So in this example, you have Feature;1 - let's say Feature is the Kingdom and 1 is the Phylum... that means the Phylum classification would be retained (since the Kingdom's classification would be implied)?

Reply:
Yep, you nailed it.